pkg/controllers/bundle/controller.go

```diff
@@ -101,7 +101,10 @@ func (h *handler) OnClusterChange(_ string, cluster *fleet.Cluster) (*fleet.Clus
 		}
 		for _, bundleDeployment := range bundleDeployments {
 			logrus.Debugf("cleaning up bundleDeployment %v in namespace %v not matching the cluster: %v", bundleDeployment.Name, bundleDeployment.Namespace, cluster.Name)
-			h.bundleDeployments.Delete(bundleDeployment.Namespace, bundleDeployment.Name, nil)
+			err := h.bundleDeployments.Delete(bundleDeployment.Namespace, bundleDeployment.Name, nil)
```

Comment:
Hm, `_ := h.bundleDeployments.Delete` is closer to the original behavior. Maybe this error is not to be checked and it is ok for bundles to not delete... 🤷 

Nah, better to check this and see if tests pass.

Reply:
`h.bundleDeployments.Delete` is a wrapper around client go delete so I suppose we properly want to handle errors. If this change results in issues we should fix them.